src/ucx_plugin.c

```diff
@@ -212,15 +214,58 @@ static void recv_handler_nbx(void *request, ucs_status_t status,
 static union ncclSocketAddress nccl_ucx_if_addr;
 static char if_name[MAX_IF_NAME_SIZE];
 
+static ncclResult_t ucx_config_no_cuda(ucp_config_t *config) {
+  char tmp[PATH_MAX];
+  const char *ucx_tls;
+  ssize_t n;
+
+  ucx_tls = getenv("NCCL_UCX_TLS");
+  if (ucx_tls == NULL) {
+    ucx_tls = getenv("UCX_TLS");
+  }
+
+  if (ucx_tls == NULL) {
+    ucx_tls = "^cuda";
+  } else if (ucx_tls[0] == '^') {
+    /* Negative expression, make sure to keep cuda excluded */
+    n = snprintf(tmp, sizeof(tmp), "^cuda,%s", &ucx_tls[1]);
+    if (n >= sizeof(tmp)) {
+      return ncclInternalError;
+    }
+
+    ucx_tls = tmp;
+  } else {
+    /* Positive expression cannot allow cuda-like transports */
+    if ((strstr(ucx_tls, "cuda") != NULL) || (strstr(ucx_tls, "gdr") != NULL)) {
+      WARN("Cannot use cuda/gdr transports as part of specified UCX_TLS");
+      return ncclInternalError;
+    }
+  }
+
+  UCXCHECK(ucp_config_modify(config, "TLS", ucx_tls));
+  UCXCHECK(ucp_config_modify(config, "RNDV_THRESH", "0"));
```

Comment:
btw, i think we also have to disable am-data scheme in RNDV protocols, i.e. set `UCX_RNDV_SCHEME=get_zcopy,put_zcopy`

Reply:
added "RNDV_SCHEME=get_zcopy" as the configuration does not take a list.

Reply:
fixed with `get_zcopy` as only one parameter can be specified